src/String.cpp

```diff
@@ -1573,6 +1573,24 @@ const char * String::utf8_str(hx::IStringAlloc *inBuffer,bool throwInvalid) cons
    return __s;
 }
 
+const char * String::utf8_str_len(hx::IStringAlloc *inBuffer,int *byteLength,bool throwInvalid) const
+{
+#ifdef HX_SMART_STRINGS
+   if (isUTF16Encoded())
+   {
+      return TConvertToUTF8(__w, byteLength, inBuffer, throwInvalid);
+   }
+   else
+   {
+      *byteLength = length;
+      return __s;
+   }
+#else
+   *byteLength = length;

```

Comment:
It might be wise to keep consistency with `TConvertToUTF8` and check that `byteLength` isn't NULL.

Reply:
Good catch, should now be wrapped in a nullptr if check.